test/unit/Cardano/Wallet/BinarySpec.hs

```diff
@@ -0,0 +1,118 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.BinarySpec (spec) where
+
+import Prelude
+
+import qualified Data.Set                         as Set
+
+import           Data.ByteArray.Encoding          (convertFromBase, Base (Base16))
+import           Data.ByteString                  (ByteString)
+import qualified Data.ByteString.Lazy.Char8 as L8
+import           Data.ByteString.Base58           (bitcoinAlphabet, decodeBase58)
+
+import Test.Hspec
+    ( Spec, describe, it, shouldBe )
+
+import Cardano.Wallet.Primitive
+import Cardano.Wallet.Binary.Helpers (unsafeDeserialiseFromBytes)
+import Cardano.Wallet.Binary (decodeBlock, decodeBlockHeader)
+
+-- A mainnet block
+block1 :: Block
+block1 = Block
+    { header = BlockHeader
+        { epochIndex = 105
+        , slotNumber = 9519
+        , prevBlockHash = hash16 "4d97da40fb62bec847d6123762e82f9325f11d0c8e89deee0c7dbb598ed5f0cf"
+        }
+    , transactions = mempty
+    }
+
+-- A mainnet block with a transaction
+block2 :: Block
+block2 = Block
+    { header = BlockHeader
+        { epochIndex = 105
+        , slotNumber = 9876
+        , prevBlockHash = hash16 "da73001193ab3e6a43921385941c5f96b8f56de3908e78fae06f038b91dadd9d"
+        }
+    , transactions = Set.fromList
+        [ Tx { inputs =
+               [ TxIn { txId = hash16 "60dbb2679ee920540c18195a3d92ee9be50aee6ed5f891d92d51db8a76b02cd2"
+                      , txIx = 3} ]
+             , outputs =
+               [ TxOut { address = addr58 "DdzFFzCqrhsug8jKBMV5Cr94hKY4DrbJtkUpqptoGEkovR2QSkcAcRgjnUyegE689qBX6b2kyxyNvCL6mfqiarzRB9TRq8zwJphR31pr"
+                       , coin = Coin 285000000
+                       }
+               , TxOut { address = addr58 "DdzFFzCqrhsmxmuQpgjUrvRwF5ZKnyQ7pGrS4q53u5B516wcc26maHz9M4myYAkQVc5m9E4DKJjRDjPxuDdK3ZsHb1Dnqf3XorZ1PnzX"
+                       , coin = Coin 1810771919
+                       }
+               ]
+             } ]
```

Comment:
Formatting is a bit "weird" here, and relies on variable-length indentation. What about: 

```suggestion
[ Tx 
    { inputs = 
        [ TxIn 
            { txId = hash16 "60dbb2679ee920540c18195a3d92ee9be50aee6ed5f891d92d51db8a76b02cd2"
            , txIx = 3
            } 
        ]
    , outputs =
        [ TxOut 
            { address = addr58 "DdzFFzCqrhsug8jKBMV5Cr94hKY4DrbJtkUpqptoGEkovR2QSkcAcRgjnUyegE689qBX6b2kyxyNvCL6mfqiarzRB9TRq8zwJphR31pr"
            , coin = Coin 285000000
            }
        , TxOut 
            { address = addr58 "DdzFFzCqrhsmxmuQpgjUrvRwF5ZKnyQ7pGrS4q53u5B516wcc26maHz9M4myYAkQVc5m9E4DKJjRDjPxuDdK3ZsHb1Dnqf3XorZ1PnzX"
            , coin = Coin 1810771919
            }
        ]
    } 
]
```
```

Reply:
Fixed